common/src/main/java/io/netty/util/concurrent/GlobalEventExecutor.java

```diff
@@ -104,15 +104,11 @@ Runnable takeTask() {
     }
 
     private void fetchFromScheduledTaskQueue() {
-        if (hasScheduledTasks()) {
-            long nanoTime = AbstractScheduledEventExecutor.nanoTime();
-            for (;;) {
-                Runnable scheduledTask = pollScheduledTask(nanoTime);
-                if (scheduledTask == null) {
-                    break;
-                }
-                taskQueue.add(scheduledTask);
-            }
+        long nanoTime = AbstractScheduledEventExecutor.nanoTime();
+        Runnable scheduledTask = pollScheduledTask(nanoTime);
+        while (scheduledTask != null) {
+            taskQueue.add(scheduledTask);
```

Comment:
@tbrooks8 @Scottmitch yeah I think we only allow to set a max capacity when using the SingleThreadEventExecutor, and so we only need to care about it there.